src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -176,15 +225,33 @@ class ArnoldOutput : public IECore::RefCounted
 				AiNodeSetStr( m_driver.get(), AiParamGetName( fileNameParameter ), output->getName().c_str() );
 			}
 
+			IECore::StringVectorDataPtr customAttributesData = new IECore::StringVectorData();
+			std::vector<std::string> &customAttributes = customAttributesData->writable();
 			for( IECore::CompoundDataMap::const_iterator it = output->parameters().begin(), eIt = output->parameters().end(); it != eIt; ++it )
 			{
 				if( boost::starts_with( it->first.string(), "filter" ) )
 				{
 					continue;
 				}
+
+				if( boost::starts_with( it->first.string(), "header:" ) )
+				{
+					std::string formattedString = formatHeaderParameter( it->first.string().substr( 7 ), it->second.get() );
+					if( !formattedString.empty())
+					{
+						customAttributes.push_back( formattedString );
+					}
+					continue;
+				}
+
 				ParameterAlgo::setParameter( m_driver.get(), it->first.c_str(), it->second.get() );
 			}
 
+			if( AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( m_driver.get() ), "custom_attributes" ) )
+			{
+				ParameterAlgo::setParameter( m_driver.get(), "custom_attributes", customAttributesData.get() );
```

Comment:
Hey John,

can we continue the discussion about this? We currently use the `"custom_attributes"` attribute from Jabuka so set metadata and the way I implemented it here would stomp over that. That currently means we can't release a new Gaffer until we also fix Jabuka and release both at the same time.

Would you object if we check whether that `StringVectorData` actually contains any data and only set the attribute if it does? We might also want to print a warning if any stomping over actually happens?

Thanks,

Matti.